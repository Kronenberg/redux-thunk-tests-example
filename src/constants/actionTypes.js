export const GET_USERS_PENDING = 'GET_USERS_PENDING';
export const GET_USERS_SUCCESS = 'GET_USERS_SUCCESS';
export const GET_USERS_REJECTED = 'GET_USERS_REJECTED';
export const GET_USERS_TO_INITIAL = 'GET_USERS_TO_INITIAL';

// SET PASSWORD FOR 'root'@'localhost' = PASSWORD('Amberq1w2e3r4t5');