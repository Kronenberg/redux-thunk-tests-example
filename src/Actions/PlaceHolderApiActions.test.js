// In this case it works
// import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";
// At this case with default(): MockStore is not a fucntion
const configureStore = require('redux-mock-store').default();
// ConfigureStore is not a fucntion
// const configureStore = require('redux-mock-store');

import getUsers from "./PlaceHolderApiActions";
import * as TYPES from '../constants/actionTypes';
import { getAction } from './utils/getTestActions';
import { payload } from "./__mocks__/user";

export const mockStore = configureMockStore([thunk]);
// recipient validation
describe("REDUX THUNK TESTER", () => {
    it("redux-thunk tester", async () => {

        const store = mockStore();
        store.dispatch(getUsers());

        expect(await getAction(store, TYPES.GET_USERS_PENDING))
            .toEqual({type: TYPES.GET_USERS_PENDING });

        expect(await getAction(store, TYPES.GET_USERS_SUCCESS))
            .toEqual({type: TYPES.GET_USERS_SUCCESS, payload});

    });

    it('async-thunk snapshot', () => {
        const store = mockStore();

        store.dispatch(getUsers());

        expect(store.getActions()).toMatchSnapshot();
    })
});