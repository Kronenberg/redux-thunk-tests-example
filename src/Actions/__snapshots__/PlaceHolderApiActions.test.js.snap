// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`REDUX THUNK TESTER async-thunk snapshot 1`] = `
Array [
  Object {
    "type": "GET_USERS_PENDING",
  },
]
`;
